app/models/concerns/searchable.rb

```diff
@@ -0,0 +1,21 @@
+module Searchable
+  extend ActiveSupport::Concern
+
+  included do
+    def self.search_by_criteria(criteria)
+      scope = self
+      scope = scope.search(criteria[:search]) if criteria[:search]
+      scope.order("#{sort_column(criteria[:sort])} #{sort_direction(criteria[:direction])}").page(criteria[:page])
+    end
+
+    private
+
+    def self.sort_column(column)
+      column_names.include?(column) ? column : 'created_at'
+    end
+
+    def self.sort_direction(direction)
+      %w[asc desc].include?(direction) ? direction : 'asc'
+    end
+  end
```

Comment:
FYI, if you want to define Class methods using `ActiveSupport::Concern`, the standard pattern is:

```ruby
module Searchable
  extend ActiveSupport::Concern

  module ClassMethods
    def search_by_criteria(criteria)
      # ...
    end
  end
end
```

See the `Permalinkable` module for an example.